x/upgrade/client/cli/query.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/upgrade"
+	"github.com/spf13/cobra"
+)
+
+// GetPlanCmd returns the query upgrade plan command
+func GetPlanCmd(storeName string, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "plan",
+		Short: "get upgrade plan (if one exists)",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			// ignore height for now
+			res, _, err := cliCtx.QueryStore([]byte(upgrade.PlanKey), storeName)
+			if err != nil {
+				return err
+			}
+
+			if len(res) == 0 {
+				return fmt.Errorf("no upgrade scheduled")
+			}
+
+			var plan upgrade.Plan
+			err = cdc.UnmarshalBinaryBare(res, &plan)
+			if err != nil {
+				return err
+			}
+			return cliCtx.PrintOutput(plan)
+		},
+	}
+}
+
+// GetAppliedHeightCmd returns the height at which a completed upgrade was applied
+func GetAppliedHeightCmd(storeName string, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "applied-height [upgrade-name]",
+		Short: "get the height at which a completed upgrade was applied",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			name := args[0]
+
+			// ignore height for now
+			res, _, err := cliCtx.QueryStore(upgrade.DoneHeightKey(name), storeName)
+			if err != nil {
+				return err
+			}
+
+			if len(res) == 0 {
+				return fmt.Errorf("no upgrade found")
+			}
+
+			var height int64
+			err = cdc.UnmarshalBinaryBare(res, &height)
```

Comment:
This needs to be the same format we serialized the data with.
Are you suggesting that https://github.com/cosmos/cosmos-sdk/blob/d5ec8427e2254fdf6e4ceec1f7ad9503b04646f7/x/upgrade/keeper.go#L139 should be updated to store it in BigEndian as well? (Cuz just changing it here would break stuff)

Reply:
I recall we discussed this with @alexanderbez like a month ago. All should use BigEndian

Reply:
resolved

Reply:
Using BigEndian for keys is super important for easy range iterations (it provides lexographical ordering). For the values, I don't see that either format has a big benefit over the other, but I agree with standardization. If BigEndian is the agreed approach in other modules, I am happy to follow.